addons/website_quote/models/sale_order.py

```diff
@@ -0,0 +1,361 @@
+            self.product_uom_id = product.uom_id.id
+            self.website_description = product and (product.quote_description or product.website_description) or ''
+            domain = {'product_uom_id': [('category_id', '=', product.uom_id.category_id.id)]}
+            return {'domain': domain}
+
+    @api.onchange('product_uom_id')
+    def product_uom_change(self):
+        if self.product_id and self.product_uom_id:
+            self.price_unit = self.product_id.uom_id._compute_price(self.product_id.uom_id.id, self.product_id.lst_price, self.product_uom_id.id)
+
+    @api.multi
+    def _inject_quote_description(self, values):
+        values = dict(values or {})
+        if not values.get('website_description') and values.get('product_id'):
+            product = self.env['product.product'].browse(values['product_id'])
+            values['website_description'] = product.quote_description or product.website_description or ''
+        return values
+
+    @api.model
+    def create(self, values):
+        values = self._inject_quote_description(values)
+        return super(SaleQuoteLine, self).create(values)
+
+    @api.multi
+    def write(self, values):
+        values = self._inject_quote_description(values)
+        return super(SaleQuoteLine, self).write(values)
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+    _description = "Sales Order Line"
+
+    website_description = fields.Html('Line Description')
+    option_line_id = fields.One2many('sale.order.option', 'line_id', 'Optional Products Lines')
+
+    @api.multi
+    def _inject_quote_description(self, values):
+        values = dict(values or {})
+        if not values.get('website_description') and values.get('product_id'):
+            product = self.env['product.product'].browse(values['product_id'])
+            values['website_description'] = product.quote_description or product.website_description
+        return values
+
+    @api.model
+    def create(self, values):
+        values = self._inject_quote_description(values)
+        return super(SaleOrderLine, self).create(values)
+
+    @api.multi
+    def write(self, values):
+        values = self._inject_quote_description(values)
+        return super(SaleOrderLine, self).write(values)
+
+    # Take the description on the order template if the product is present in it
+    @api.onchange('product_id')
+    def product_id_change(self):
+        domain = super(SaleOrderLine, self).product_id_change()
+        if self.order_id.template_id:
+            self.name = next((quote_line.name for quote_line in self.order_id.template_id.quote_line if quote_line.product_id.id == self.product_id.id), self.name)
+        return domain
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _default_template_id(self):
+        template = self.env.ref('website_quote.website_quote_template_default', raise_if_not_found=False)
+        return template or False
```

Comment:
`or False` not required. If not exist, `template` will be `None` and it will work too.